context.go

```diff
@@ -439,14 +439,13 @@ func (c *context) JSON(code int, i interface{}) (err error) {
 		indent string
 	)
 	if _, pretty = c.QueryParams()["pretty"]; c.echo.Debug || pretty {
-		pretty = true
 		indent = "  "
 	}
-	return c.jsonBlob(code, i, pretty, indent)
+	return c.jsonBlob(code, i, indent)
 }
 
 func (c *context) JSONPretty(code int, i interface{}, indent string) (err error) {
-	return c.jsonBlob(code, i, true, indent)
+	return c.jsonBlob(code, i, indent)
```

Comment:
This looks risky. It works fine in `JSON()` where we control the indent ourselves, here however it messes things up :( If they want it pretty but with a blank indent (which I have no idea what it would do? just put each field on a separate line with no indent?) they will get it "not pretty". Should we return an error when the indent is blank for pretty? and also update the documentation to reflect that we expect a non blank indent to be provided for this function? 

Yes, just checked, that's what it currently does, with a blank indent, it prints pretty with no indent:

```JSON
{
"bar": 1,
"foo": 42
}
```

We can either revert back to using `bool` OR we can convert `indent` in `jsonBlob()` to be a pointer to string. That way, we don't check if it is blank, we check if it is nil. If nil == ugly json, if present (even if blank) == pretty json with that ident. That would allow us to keep the existing behavior where we accept even a blank indent.

Reply:
yeah, you'll right

Reply:
fixed with `indent` pointer

Reply:
Looks good to me :+1:

Reply:
Glad to hear.. waiting for @vishr review